Dockerfile

```diff
@@ -1,173 +1,52 @@
-RUN npm ci
-COPY ./src/extensions/hooks/gp-tokens .
-RUN npm run build
-
-# Build hooks/directus-users
-FROM node:18-alpine AS builder-17
-WORKDIR /builder/src/extensions/hooks/directus-users
-ADD ./src/extensions/hooks/directus-users/package.json ./src/extensions/hooks/directus-users/package-lock.json ./
-RUN npm ci
-COPY ./src/extensions/hooks/directus-users .
-RUN npm run build
-
-# Build hooks/location-verrides
-FROM node:18-alpine AS builder-18
-WORKDIR /builder/src/extensions/lib
-ADD ./src/extensions/lib/package.json ./src/extensions/lib/package-lock.json ./
-RUN npm ci
-COPY ./src/extensions/lib .
-WORKDIR /builder/src/extensions/hooks/location-overrides
-ADD ./src/extensions/hooks/location-overrides/package.json ./src/extensions/hooks/location-overrides/package-lock.json ./
-RUN npm ci
-COPY ./src/extensions/hooks/location-overrides .
-RUN npm run build
+FROM node:18-alpine as builder
+RUN corepack enable
+WORKDIR /builder
+COPY package.json pnpm-*.yaml ./
+
+# START: generate via scripts/docker-ls.sh
+COPY src/extensions/lib/package.json src/extensions/endpoints/adoption-code/package.json ./
+COPY src/extensions/endpoints/sync-github-data/package.json src/extensions/endpoints/sync-github-data/
+COPY src/extensions/hooks/adopted-probe/package.json src/extensions/hooks/adopted-probe/
+COPY src/extensions/hooks/directus-users/package.json src/extensions/hooks/directus-users/
+COPY src/extensions/hooks/gp-tokens/package.json src/extensions/hooks/gp-tokens/
+COPY src/extensions/hooks/location-overrides/package.json src/extensions/hooks/location-overrides/
+COPY src/extensions/hooks/sign-in/package.json src/extensions/hooks/sign-in/
+COPY src/extensions/hooks/sign-up/package.json src/extensions/hooks/sign-up/
+COPY src/extensions/interfaces/github-username/package.json src/extensions/interfaces/github-username/
+COPY src/extensions/interfaces/gp-tags/package.json src/extensions/interfaces/gp-tags/
+COPY src/extensions/interfaces/token/package.json src/extensions/interfaces/token/
+COPY src/extensions/lib/package.json src/extensions/lib/
+COPY src/extensions/modules/probes-adapter/package.json src/extensions/modules/probes-adapter/
+COPY src/extensions/operations/adopted-probes-credits-cron-handler/package.json src/extensions/operations/adopted-probes-credits-cron-handler/
+COPY src/extensions/operations/adopted-probes-status-cron-handler/package.json src/extensions/operations/adopted-probes-status-cron-handler/
+COPY src/extensions/operations/gh-webhook-handler/package.json src/extensions/operations/gh-webhook-handler/
+COPY src/extensions/operations/remove-banned-users-cron-handler/package.json src/extensions/operations/remove-banned-users-cron-handler/
+COPY src/extensions/operations/sponsors-cron-handler/package.json src/extensions/operations/sponsors-cron-handler/
+COPY src/extensions/token-value/package.json src/extensions/token-value/
+# END: generate via scripts/docker-ls.sh
+
+RUN pnpm install
```

Comment:
shouldn't here be `pnpm install --frozen-lockfile`?

Reply:
The main reason for using `ci` is that it errors if the lock file is missing, but pnpm doesn't do that, so it isn't very useful. Also, it already defaults to this flag being true in CI environments 😄